utils/neural_magic/sparse_train_manager.py

```diff
@@ -191,6 +194,49 @@ def turn_off_scaler(self, scaler: torch.cuda.amp.GradScaler):
         """
         scaler._enabled = False
 
+    def rescale_gradient_accumulation(
+        self, batch_size: int, accumulate: int, image_size: int
+    ) -> Tuple[int, int]:
+        """
+        Used when autobatch and QAT are both enabled. Training with QAT adds additional
+        overhead which can cause OOM errors if autobatch is enabled. This function
+        rescales batch size and gradient accumulation to fit into memory with QAT while
+        maintaining the original effective batch size
+        """
+        # Temporary copy of the model with QAT applied
+        quant_model_copy = deepcopy(de_parallel(self.model))
```

Comment:
Concerned this could cause OOM issues. Could we apply QAT to the original model before the modifier does or is that likely to cause issues?

Reply:
As long as it only happens once (and maybe before training loop?) probably not an issue